presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsMetadata.java

```diff
@@ -0,0 +1,375 @@
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.SchemaTablePrefix;
+import com.facebook.presto.spi.SortingProperty;
+import com.facebook.presto.spi.block.SortOrder;
+import com.facebook.presto.spi.connector.ConnectorMetadata;
+import com.facebook.presto.spi.predicate.TupleDomain;
+import com.facebook.presto.spi.type.BigintType;
+import com.facebook.presto.spi.type.DateType;
+import com.facebook.presto.spi.type.IntegerType;
+import com.facebook.presto.spi.type.TimeType;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.teradata.tpcds.Table;
+import com.teradata.tpcds.column.CallCenterColumn;
+import com.teradata.tpcds.column.CatalogPageColumn;
+import com.teradata.tpcds.column.CatalogReturnsColumn;
+import com.teradata.tpcds.column.CatalogSalesColumn;
+import com.teradata.tpcds.column.Column;
+import com.teradata.tpcds.column.ColumnType;
+import com.teradata.tpcds.column.CustomerAddressColumn;
+import com.teradata.tpcds.column.CustomerColumn;
+import com.teradata.tpcds.column.CustomerDemographicsColumn;
+import com.teradata.tpcds.column.DateDimColumn;
+import com.teradata.tpcds.column.HouseholdDemographicsColumn;
+import com.teradata.tpcds.column.IncomeBandColumn;
+import com.teradata.tpcds.column.InventoryColumn;
+import com.teradata.tpcds.column.ItemColumn;
+import com.teradata.tpcds.column.PromotionColumn;
+import com.teradata.tpcds.column.ReasonColumn;
+import com.teradata.tpcds.column.ShipModeColumn;
+import com.teradata.tpcds.column.StoreColumn;
+import com.teradata.tpcds.column.StoreReturnsColumn;
+import com.teradata.tpcds.column.StoreSalesColumn;
+import com.teradata.tpcds.column.TimeDimColumn;
+import com.teradata.tpcds.column.WarehouseColumn;
+import com.teradata.tpcds.column.WebPageColumn;
+import com.teradata.tpcds.column.WebReturnsColumn;
+import com.teradata.tpcds.column.WebSalesColumn;
+import com.teradata.tpcds.column.WebSiteColumn;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
+import static com.facebook.presto.spi.type.VarcharType.createVarcharType;
+import static com.facebook.presto.tpcds.Types.checkType;
+import static java.util.Objects.requireNonNull;
+
+public class TpcdsMetadata
+        implements ConnectorMetadata
+{
+    public static final String TINY_SCHEMA_NAME = "tiny";
+    public static final double TINY_SCALE_FACTOR = 0.01;
+
+    public static final List<String> SCHEMA_NAMES = ImmutableList.of(
+            TINY_SCHEMA_NAME, "sf1", "sf10", "sf100", "sf300", "sf1000", "sf3000", "sf10000", "sf30000", "sf100000");
+
+    private final Set<String> tableNames;
+
+    public TpcdsMetadata()
+    {
+        ImmutableSet.Builder<String> tableNames = ImmutableSet.builder();
```

Comment:
Converted the loop to a stream with the proper Locale.

The tpcds generator can be found here: https://github.com/Teradata/tpcds.